### PR TITLE
Let identity version of overapproximate accept kwargs

### PR DESCRIPTION
**Pull request by schillic**
See [this comment](https://github.com/JuliaReach/Reachability.jl/pull/641#discussion_r299186663).

### CODE REVIEW

File: src/Approximations/overapproximate.jl [hunk 1]
```diff
@@ -1,19 +1,20 @@
 using LazySets: block_to_dimension_indices, substitute_blocks, get_constrained_lowdimset
 """
-    overapproximate(X::S, ::Type{S}) where {S<:LazySet}
+    overapproximate(X::S, ::Type{S}, kwargs...) where {S<:LazySet}
```
Comment (mforets) [inline src/Approximations/overapproximate.jl]:
```suggestion
    overapproximate(X::S, ::Type{S}; kwargs...) where {S<:LazySet}
```

File: src/Approximations/overapproximate.jl [hunk 2]
```diff
@@ -1,19 +1,20 @@
...
+- `X`       -- set
+- `Type{S}` -- target set type
+- `kwargs`  -- further keyword arguments (ignored)
 
 ### Output
 
 The input set.
 """
-function overapproximate(X::S, ::Type{S}) where {S<:LazySet}
+function overapproximate(X::S, ::Type{S}, kwargs...) where {S<:LazySet}
```
Comment (mforets) [inline src/Approximations/overapproximate.jl]:
```suggestion
function overapproximate(X::S, ::Type{S}; kwargs...) where {S<:LazySet}
```

Comment (schillic) [reply src/Approximations/overapproximate.jl]: This does not work as I intended it:
```julia
julia> X = rand(VPolygon)
overapproxiVPolygon{Float64,Array{Float64,1}}(maArray{Float64,1}[[1.33152, -0.716187], [3.00489, 0.152034], [3.87972, 1.08194]])

julia> overapproximate(X, VPolygon)  # good
VPolygon{Float64,Array{Float64,1}}(Array{Float64,1}[[1.33152, -0.716187], [3.00489, 0.152034], [3.87972, 1.08194]])

julia> overapproximate(X, VPolygon, Hyperrectangle)  # bad
ERROR: MethodError: no method matching overapproximate(::VPolygon{Float64,Array{Float64,1}}, ::Type{VPolygon}, ::Type{Hyperrectangle})
Closest candidates are:
  overapproximate(::S<:LazySet, ::Type{S<:LazySet}; kwargs...) where S<:LazySet at LazySets/src/Approximations/overapproximate.jl:18
  overapproximate(::LazySet{N<:Real}, ::Type{#s62} where #s62<:HPolygon) where N<:Real at LazySets/src/Approximations/overapproximate.jl:44
  overapproximate(::LazySet{N<:Real}, ::Type{#s52} where #s52<:HPolygon, ::Real) where N<:Real at LazySets/src/Approximations/overapproximate.jl:44
  ...
```

Comment (mforets) [reply src/Approximations/overapproximate.jl]: Ok then this wants to allow additional arguments, not keyword arguments,

```julia
function overapproximate(X::S, ::Type{S}, args...) where {S<:LazySet}
```

File: src/Approximations/overapproximate.jl [hunk 3]
```diff
@@ -1,19 +1,20 @@
...
 
 Overapproximating a set of type `S` with type `S` is a no-op.
 
 ### Input
 
-- `X` -- set
-- `Type{S}` -- set type
+- `X`       -- set
+- `Type{S}` -- target set type
+- `kwargs`  -- further keyword arguments (ignored)
```
Comment (mforets) [inline src/Approximations/overapproximate.jl]:
```suggestion
- `args`  -- further arguments (ignored)
```

File: src/Approximations/overapproximate.jl (hunk 2 above)
Comment (mforets) [reply src/Approximations/overapproximate.jl]: Shall we add this to the unit tests:

```julia
oa = overapproximate(X, VPolygon, Hyperrectangle)  # the third argument is ignored
@test oa isa VPolygon
```

Comment (schillic) [reply src/Approximations/overapproximate.jl]: Hm, from the other usages of the function, it should be clear that the output is a `VPolygon`. The test would cover that you can add a third argument, though, so I will add it :+1:

**Pull request merged**